src/vm/amd64/umthunkstub.S

```diff
@@ -0,0 +1,368 @@
+        movdqa          xmm2, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 20h]
+        movdqa          xmm3, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 30h]
+        movdqa          xmm4, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 40h]
+        movdqa          xmm5, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 50h]
+        movdqa          xmm6, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 60h]
+        movdqa          xmm7, [rsp + TheUMEntryPrestub_XMM_SAVE_OFFSET + 70h]
+
+        //
+        // epilogue
+        //
+        add             rsp, TheUMEntryPrestub_STACK_FRAME_SIZE
+        TAILJMP_RAX
+        
+NESTED_END TheUMEntryPrestub, _TEXT
+
+
+//
+// METHODDESC_REGISTER: UMEntryThunk*
+//
+NESTED_ENTRY UMThunkStub, _TEXT, UMThunkStubUnwindFrameChainHandler
+
+// number of integer registers saved in prologue
+#define UMThunkStubAMD64_NUM_REG_PUSHES 2
+
+#define UMThunkStubAMD64_METHODDESC_SPILL_SIZE 0x10
+#define UMThunkStubAMD64_METHODDESC_SPILL_NEGOFFSET 0x20
+
+// HOST_NOTIFY_FLAG
+#define UMThunkStubAMD64_HOST_NOTIFY_FLAG_NEGOFFSET 0x28
+
+// XMM save area 
+#define UMThunkStubAMD64_XMM_SAVE_NEGOFFSET 0xa8
+
+// Add in the callee scratch area size.
+#define UMThunkStubAMD64_CALLEE_SCRATCH_SIZE SIZEOF_MAX_OUTGOING_ARGUMENT_HOMES
+#define UMThunkStubAMD64_STACK_FRAME_SIZE 0xd8
+
+// Now we have the full size of the stack frame.  The offsets have been computed relative to the
+// top, so negate them to make them relative to the post-prologue rsp.
+#define UMThunkStubAMD64_FRAME_OFFSET UMThunkStubAMD64_CALLEE_SCRATCH_SIZE
+#define UMThunkStubAMD64_METHODDESC_SPILL_OFFSET (UMThunkStubAMD64_STACK_FRAME_SIZE - UMThunkStubAMD64_FRAME_OFFSET - UMThunkStubAMD64_METHODDESC_SPILL_NEGOFFSET)
+#define UMThunkStubAMD64_HOST_NOTIFY_FLAG_OFFSET (UMThunkStubAMD64_STACK_FRAME_SIZE - UMThunkStubAMD64_FRAME_OFFSET - UMThunkStubAMD64_HOST_NOTIFY_FLAG_NEGOFFSET)
+#define UMThunkStubAMD64_XMM_SAVE_OFFSET (UMThunkStubAMD64_STACK_FRAME_SIZE - UMThunkStubAMD64_FRAME_OFFSET - UMThunkStubAMD64_XMM_SAVE_NEGOFFSET)
+#define UMThunkStubAMD64_ARGUMENTS_STACK_HOME_OFFSET (UMThunkStubAMD64_STACK_FRAME_SIZE + 8 - UMThunkStubAMD64_FRAME_OFFSET)
+#define UMThunkStubAMD64_FIXED_STACK_ALLOC_SIZE (UMThunkStubAMD64_STACK_FRAME_SIZE - (UMThunkStubAMD64_NUM_REG_PUSHES * 8))
+
+#if (UMTHUNKSTUB_HOST_NOTIFY_FLAG_RBPOFFSET - UMThunkStubAMD64_HOST_NOTIFY_FLAG_OFFSET) != 0
+#error update UMTHUNKSTUB_HOST_NOTIFY_FLAG_RBPOFFSET
+#endif
+
+        push_nonvol_reg r12
+        push_nonvol_reg rbp                                                                     // stack_args
+        alloc_stack     UMThunkStubAMD64_FIXED_STACK_ALLOC_SIZE
+        set_frame       rbp, UMThunkStubAMD64_FRAME_OFFSET                                      // stack_args
+        mov             byte ptr [rbp + UMThunkStubAMD64_HOST_NOTIFY_FLAG_OFFSET], 0            // hosted
+
+        mov             [rbp + UMThunkStubAMD64_ARGUMENTS_STACK_HOME_OFFSET +  0h], rdi
```

Comment:
This has the same problem as TheUMEntryPrestub - there are no outgoing argument homes on Unix.